apps/files/src/components/OcCollaborationStatus.vue

```diff
@@ -0,0 +1,199 @@
+                    <span v-if="c.role && c.expires">|</span>
+                    <span v-if="c.expires">Expires: {{ c.expires }}</span>
+                  </span>
+                </template>
+                <template slot="actions">
+                  <oc-icon name="delete" @click="onDelete(c)" />
+                  <oc-icon v-if="editing != c" name="edit" @click="edit(c)" />
+                </template>
+              </oc-user>
+
+              <div v-if="editing == c" class="uk-margin-small-top">
+                <div v-for="(role, r) in roles" :key="r" @click="c.role = r">
+                  <div class="uk-inline">
+                    <oc-radio :model="c.role" :value="r" />
+                  </div>
+                  <div class="uk-inline">
+                    <div>
+                      <strong v-text="role.name" />
+                    </div>
+                    <div v-text="role.description" />
+                  </div>
+                </div>
+                <div class="uk-container uk-padding-remove uk-margin-small-top">
+                  <oc-button icon="save" class="uk-align-right" text="Save" @click="save(c)" />
+                </div>
+              </div>
+            </li>
+          </ul>
+        </template>
+      </oc-accordion-item>
+    </oc-accordion>
+  </div>
+</template>
+
+<script>
+/**
+ * The collaboration status shows
+ *
+ * * the current owner,
+ * * an input field to search for new collaboratores,
+ * * the list of current collaborators and their role
+ *
+ * It needs to be filled with an owner and an array of collaborators.
+ * The roles can be overwritten as well.
+ *
+ * ##TODO:
+ * - [ ] add search for users autocomplete component
+ * - [ ] use mixin to format mtime, would add dependency to moment, can be done in phoenix
+ */
+export default {
+  name: 'oc-collaboration-status',
+  status: 'review',
+  release: '1.0.0',
+  // passed in
+  props: {
+    /**
+     * Owner, a user object
+     **/
+    owner: {
+      type: Object,
+      required: true
+    },
+    /**
+     * Roles
+     **/
+    roles: {
+      type: Object,
+      required: true
+    },
+    /**
+     * user objects with a `role` and `expires` property
+     **/
+    collaborators: {
+      type: Array,
+      required: true
+    }
+  },
+  data () {
+    return {
+      editing: null
+    }
+  },
+  methods: {
+    edit (share) {
+      this.editing = share
+    },
+    save (collaborator) {
+      /**
+       * Save event
+       * @event save
+       * @type {collaborator}
+       */
+      this.$emit('save', collaborator)
+      this.editing = null
+    },
+    onDelete (collaborator) {
+      /**
+       * delete cellaborator event
+       * @event delete
+       * @type {collaborator}
+       */
+      this.$emit('delete', collaborator)
+      /** TODO would mutate property
+      this.shares = this.shares.filter(function (s){
+        return s != share
+      })
+      */
+    }
+  }
+}
+</script>
+
+<docs>
```

Comment:
no docs needed in here

Reply:
I left them in to see what kind of objects need to be passed in as props. Could be removed after logic has been added ... then again ... an example to use this component is good to have, isn't it?

Reply:
I see no need - the component will be used as is within phoenix - this will change anyhow over time.